src/code42cli/logger/handlers.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import socket
+import ssl
+from logging.handlers import SysLogHandler
+
+from code42cli.logger.enums import ServerProtocol
+
+
+class NoPrioritySysLogHandler(SysLogHandler):
+    """
+    Overrides the default implementation of SysLogHandler to not send a `<PRI>` at the
+    beginning of the message. Most CEF consumers seem to not expect the `<PRI>` to be
+    present in CEF messages. Attach to a logger via `.addHandler` to use.
+
+    `self.socket` is lazily loaded for testing purposes, so the connection does not get
+    made for TCP/TLS until the first log record is about to be transmitted.
+
+    Args:
+        hostname: The hostname of the syslog server to send log messages to.
+        port: The port of the syslog server to send log messages to.
+        protocol: The protocol over which to submit syslog messages. Accepts TCP, UDP, or TLS.
+    """
+
+    def __init__(self, hostname, port, protocol, certs):
+        self._hostname = hostname
+        self._port = port
+        self._certs = certs
+        self.address = (hostname, port)
+        logging.Handler.__init__(self)
+        self._use_insecure = protocol != ServerProtocol.TLS_TCP
+        self.socktype = _try_get_socket_type_from_protocol(protocol)
+        self.socket = None
+
+    def connect_socket(self):
```

Comment:
So I couldn't really make this method internal as @alanag13  suggested because I want this to get called in the `get_logger()` method so that we can test the connection.. I don't want it initialize in the init method because it makes some of the testing require less mocks (And also principally it is better to not  make a network request in an initializer). I hope these changes make it more obvious.. See the `get_logger_for_server` method as well.